src/internet_identity/src/main.rs

```diff
@@ -115,6 +115,24 @@ fn lookup(anchor_number: AnchorNumber) -> Vec<DeviceData> {
     })
 }
 
+#[query]
+fn get_device_credentials(anchor_number: AnchorNumber) -> GetDeviceCredentialsResponse {
+    let anchor = state::anchor(anchor_number);
+
+    let credentials = anchor.get_device_credentials(|d| d.purpose == Purpose::Authentication);
+    let recovery_credentials = anchor.get_device_credentials(|d| d.purpose == Purpose::Recovery);
```

Comment:
Why not fetch all devices and partition them? If a new `Purpose` is introduced, we may forget to fetch it here

Reply:
TIL about `partition`. Thanks for pointing it out.